library/rsa.c

```diff
@@ -1429,15 +1442,27 @@ int mbedtls_rsa_rsassa_pkcs1_v15_verify( mbedtls_rsa_context *ctx,
     if( asn1_len != hashlen )
         return( MBEDTLS_ERR_RSA_VERIFY_FAILED );
 
-    if( memcmp( p, hash, hashlen ) != 0 )
-        return( MBEDTLS_ERR_RSA_VERIFY_FAILED );
+    ret = MBEDTLS_ERR_RSA_VERIFY_FAILED;
```

Comment:
It's to ensure `ret` has a failure value (instead of the success leftover from previous function calls), so if the `memcmp()`call  is skipped due to FI you'll have a failure.